File: Java Old Code/Class10/src/BinarySearch.java
import java.util.*;
public class BinarySearch
{
    public static void main(String args[]) 
   {
       Scanner sc = new Scanner(System.in);
       System.out.println("Enter Number of Array Elements");
       int ne = sc.nextInt();
       int arr[]= new int[ne];
       int n = arr.length;
       System.out.println("Enter Elements of the Array in Ascending Order:- ");
       for(int i=0;i<arr.length;i++)
           arr[i] = sc.nextInt(); 
       int f = 0;
       int l = arr.length-1;
       System.out.println("Enter Element You Want to search:- ");
       int num = sc.nextInt();
       while(f<=l)
       {
          int mid = (f+l)/2;
          if(num==arr[mid])
          {
              System.out.println("Search Successful");
              System.exit(0);
          }
          if(num>arr[mid])
              f = mid+1;
          else if(num<arr[mid])

              l = mid-1;      
       }
       System.out.println("Search Unsuccessful");
   }
}

File: C++_DSA/BinarySearch.cpp
#include<bits/stdc++.h>
using namespace std;

void BinarySearch(int arr[], int l, int h, int x){
    if(l>h) return;
    int mid = l + (h-l)/2;
    if(arr[mid]==x) cout<<"Element found at position "<<mid<<endl;
    else if(arr[mid]<x) BinarySearch(arr, mid+1, h, x);
    else BinarySearch(arr, l, mid-1, x);
}

int main(){
    int arr[] = {10, 12, 13, 16, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30};
    int n = sizeof(arr)/sizeof(arr[0]);
    int x = 18;
    BinarySearch(arr, 0, n-1, x);
    return 0;
}